pkg/datasources/datasource_user.go

```diff
@@ -0,0 +1,76 @@
+package datasources
+
+import (
+	"context"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/frontegg"
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func User() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: userDataSourceRead,
+
+		Description: `The user data source allows you to retrieve information about a user in your Materialize organization.`,
+
+		Schema: map[string]*schema.Schema{
+			"email": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The email address of the user to retrieve.",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique (UUID) identifier of the user.",
+			},
+			"verified": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether the user's email address has been verified.",
+			},
+			"metadata": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Additional metadata associated with the user.",
+			},
+			"auth_provider": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The authentication provider for the user.",
+			},
```

Comment:
I think this one's likely to be confusing, too. IIUC it's, like, the _last_ auth provider the user used to sign in, so it jumps around a lot.